### PR TITLE
block import time metrics

### LINKED ISSUE #1707: Build a sidechain block production metrics
### Context

Can be a complex topic.

We'd like to measure the time consumption distribution during sidechain block production for the sake of:
- statistics
- monitoring
- better understanding of the bottleneck before further optimisation

In particular, it should include but not limited to the following:
- sidechain block import
- parachain block import, including I/O
- state pre-update, including I/O
- stf execution
- block composition
- after effect
- any others?

It would be great if we could visualise it.

If done, it will supersede the #1611

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by kziemianek**
Adds more detailed metrics to montior parentchain/sidechain block import time and sidechain block production.

resolves: P-50
resolves: P-60
resolves: #1707

### CODE REVIEW

File: tee-worker/core-primitives/enclave-metrics/src/lib.rs [hunk 1]
```diff
@@ -42,7 +44,8 @@ pub enum EnclaveMetric {
 	StfTaskExecutionTime(Box<RequestType>, f64),
 	SuccessfulTrustedOperationIncrement(TrustedCall),
 	FailedTrustedOperationIncrement(TrustedCall),
-	// OracleMetric(OracleMetric<MetricsInfo>),
+	ParentchainBlockImportTime(Duration),
+	SidechainBlockImportTime(Duration), // OracleMetric(OracleMetric<MetricsInfo>),
```
Comment (Kailai-Wang) [inline tee-worker/core-primitives/enclave-metrics/src/lib.rs]: The comment is bit confusing

File: tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs [hunk 2]
```diff
@@ -23,18 +23,18 @@ use std::sync::RwLock;
...
 use codec::{Decode, Encode};
 use core::marker::PhantomData;
-use itp_ocall_api::EnclaveSidechainOCallApi;
+use itp_ocall_api::{EnclaveMetricsOCallApi, EnclaveSidechainOCallApi};
 use itp_types::{BlockHash, ShardIdentifier};
 use sgx_types::{sgx_status_t, SgxResult};
 use std::vec::Vec;
 
-pub struct SidechainOCallApiMock<SignedSidechainBlockType> {
+pub struct OCallApiMock<SignedSidechainBlockType> {
```
Comment (Kailai-Wang) [inline tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs]: Maybe we can leave it unchanged? It's not that important and we could keep it in sync with upstream

**Review by Kailai-Wang — COMMENTED** [review]
Thanks, looks good!
However, I don't think it resolves P-50 completely.

Ideal goal of P-50 is to be able to visualize the breakdown of a sidechian block frame

**Comment by kziemianek** [comment]
> Thanks, looks good! However, I don't think it resolves P-50 completely.
> 
> Ideal goal of P-50 is to be able to visualize the breakdown of a sidechian block frame

Thanks @Kailai-Wang , regarding more granular metrics we have another ticket P-60 - I've noticed it once I finished this implementation hovewer I can continue topic in this pull request.

File: tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs (hunk 2 above)
Comment (BillyWooo) [reply tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs]: I think what you mean is: try to keep sync with upstream with least effort, right? @Kailai-Wang 
I would suggest not to touch it, unless it blocks us from any development/functionality.

**Comment by kziemianek** [comment]
@Kailai-Wang can we merge this PR and continue P-60 in follow up ?

**Review by BillyWooo — COMMENTED** [review]
I would suggest let's revert the name change: `SidechainOCallApi -> OCallApi`.

File: tee-worker/core-primitives/enclave-metrics/src/lib.rs (hunk 1 above)
Comment (kziemianek) [reply tee-worker/core-primitives/enclave-metrics/src/lib.rs]: moved to next line, actually this is commented out variant

File: tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs (hunk 2 above)
Comment (kziemianek) [reply tee-worker/core-primitives/test/src/mock/sidechain_ocall_api_mock.rs]: changed

**Comment by kziemianek** [comment]
> I would suggest let's revert the name change: `SidechainOCallApi -> OCallApi`.

Renamed

**Review by BillyWooo — APPROVED** [review]
Thank you. LGTM.
Please resolve the conflicts.

**Pull request merged**